contracts/ibc_transfer/src/contract.rs

```diff
@@ -112,14 +114,18 @@ fn msg_with_sudo_callback<C: Into<CosmosMsg<T>>, T>(
 // and process this payload when an acknowledgement for the SubmitTx message is received in Sudo handler
 fn prepare_sudo_payload(mut deps: DepsMut, _env: Env, msg: Reply) -> StdResult<Response> {
     let payload = read_reply_payload(deps.storage, msg.id)?;
-    let resp: MsgIbcTransferResponse = from_json(
-        msg.result
+    let resp: MsgTransferResponse = decode_message_response(
+        &msg.result
             .into_result()
             .map_err(StdError::generic_err)?
-            .data
-            .ok_or_else(|| StdError::generic_err("no result"))?,
+            .msg_responses[0]
```

Comment:
Could you please add a comment why do we use only the first element here?

Reply:
Done